src/service/resources-impl.js

```diff
@@ -1216,11 +1215,19 @@ export class Resources {
     // the overflow callbacks are called.
     const now = Date.now();
     const viewportRect = this.viewport_.getRect();
-    const scrollHeight = this.viewport_.getScrollHeight();
     const topOffset = viewportRect.height / 10;
     const bottomOffset = viewportRect.height / 10;
-    const maxDocBottomOffset = scrollHeight - DOC_BOTTOM_OFFSET_LIMIT_;
-    const docBottomOffset = Math.max(scrollHeight * 0.85, maxDocBottomOffset);
+    /**
+     * @param {!../layout-rect.LayoutRectDef} layoutBox
+     * @param {!../layout-rect.LayoutRectDef} initialLayoutBox
+     * @returns True if element is within 15% and 1000px of document bottom.
+     */
+    const elementNearBottom = (layoutBox, initialLayoutBox) => {
```

Comment:
Pass in `viewport` and we can hoist this out of the method.

Reply:
Kept `this.viewport_` reference and hoisted anyways.